### PR TITLE
Expose `split` in dataset builders

### PR DESCRIPTION
**Pull request by SalmanMohammadi**
#### Context
What is the purpose of this PR? Is it to
- [x] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Discussed in https://github.com/pytorch/torchtune/issues/1107#issuecomment-2239651967. This exposes the `split` arg for `datasets.load_dataset` in our default dataset builders - particularly useful (as documented) for specifying fractions of datasets to train with, which all levels of users might enjoy.

#### Test plan

👁️     .    👁️

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [x] add unit tests for any new functionality
- [x] update docstrings for any new or updated methods or classes
- [x] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

File: torchtune/datasets/_chat.py [hunk 1]
```diff
@@ -30,9 +30,6 @@ class ChatDataset(Dataset):
     The general flow from loading a sample to tokenized prompt is:
     load sample -> apply transform -> foreach turn{format into template -> tokenize}
 
-    If the column/key names differ from the expected names in the :class:`~torchtune.data.ChatFormat`,
```
Comment (SalmanMohammadi) [inline torchtune/datasets/_chat.py:33]: I thought this was a vestigial doc so removed.

**Review by RdoubleA — APPROVED** [review]
Thanks for this improvement Salman!
